src/session/content/message_row/media.rs

```diff
@@ -155,24 +175,50 @@ impl Media {
     }
 
     pub(crate) fn caption(&self) -> String {
-        self.imp().caption_label.label().into()
+        self.imp()
+            .caption_label
+            .borrow()
+            .as_ref()
+            .map(|c| c.label())
+            .unwrap_or_default()
     }
 
-    pub(crate) fn set_caption(&self, caption: &str) {
+    pub(crate) fn set_caption(&self, caption: String) {
         if self.caption() == caption {
             return;
         }
 
         let imp = self.imp();
+        let indicators = self.indicators();
+
         if caption.is_empty() {
-            imp.caption_label.set_visible(false);
+            let caption_label = imp.caption_label.take().unwrap();
+            imp.content.remove(&caption_label);
+
+            // Drop the label so that it will unparent the indicators
+            drop(caption_label);
```

Comment:
Because the reference counter goes to 0, so the widget is disposed and the `dispose` of that widget unparents the indicators.

Reply:
Could you not just call `indicators.unparent();`?

Reply:
Nope. The reason is that the `caption_label` will still be dropped right after that block. So, what it will happen is that you first unparent the indicators from the label, then you set the overlay as it's new parent, but then right after that block the `MessageLabel` will be dropped and then it will call `unparent()` again on the indicators, which it will make the indicators to be out-of-tree.

Reply:
Ok, I see now. This is a bad side effect. We could prevent that generally by iterating over the children in `dispose` and unparent them like we do in some places.
This at least wouldn't violate the Principle of Least Surprise.

Reply:
I don't really like that because `MessageLabel` is actually designed to always have a `MessageIndicators` as its child. So, calling `unparent` to that widget is not really something that makes sense imho, because we always manage the indicators e.g. in the measure phase, which is specifically designed to be managed by children. I find more sensible to just drop the label altogether and just let the `dispose` to properly do all its things and to completely remove all references of the indicators in the message label before reusing it elsewhere.

Reply:
Ok, I get your point. But I would suggest making it a bit more elegant. IMHO, this is too expressive. AFAIK, this line alone
```rust
    // The label will also be dropped so that it will unparent the indicators.
    imp.content.remove(&imp.caption_label.take().unwrap());
```
should drop it implicitly.

Reference: 
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=38bad2ec63d4e19485e89e8a27036222